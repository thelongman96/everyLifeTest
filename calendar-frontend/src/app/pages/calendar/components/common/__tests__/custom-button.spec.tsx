import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import CustomButton from '../custom-button';

const onClickFuncMock = jest.fn();

const defaultProps = {
  buttonText: 'test',
  onClickFunc: onClickFuncMock,
  disabled: false,
  isLoading: false,
};

const renderCustomButton = (props = defaultProps) => {
  return render(
    <CustomButton
      buttonText={props.buttonText}
      onClickFunc={props.onClickFunc}
      disabled={props.disabled}
      isLoading={props.isLoading}
    />,
  );
};

describe('CustomButtonComponent', () => {
  it('renders correctly', () => {
    const { container } = renderCustomButton();

    expect(container).toMatchSnapshot();
  });

  describe('On Click - Calls onClickFunc', () => {
    it('should ', async () => {
      renderCustomButton();

      const btn = screen.getByTestId('test_btn');
      userEvent.click(btn);

      expect(onClickFuncMock).toHaveBeenCalled();
    });
  });

  describe('Button disabled when disabled prop is true', () => {
    it('should ', async () => {
      renderCustomButton({ ...defaultProps, disabled: true });

      const btn = screen.getByTestId('test_btn');

      expect(btn).toBeDisabled();
    });
  });
});
