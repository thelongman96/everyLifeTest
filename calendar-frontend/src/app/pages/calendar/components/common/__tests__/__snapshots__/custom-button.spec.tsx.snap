// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButtonComponent renders correctly 1`] = `
<div>
  <div
    class="customButton__container"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary customButton__button css-1uent87-MuiButtonBase-root-MuiButton-root"
      data-testid="test_btn"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-loadingWrapper"
        style="display: contents;"
      />
      test
    </button>
  </div>
</div>
`;
